subsys/mgmt/suitfu/src/suitfu_mgmt_img.c

```diff
@@ -82,24 +87,43 @@ static int suitfu_mgmt_img_upload(struct smp_streamer *ctx)
 			return MGMT_ERR_EINVAL;
 		}
 
-		suit_plat_err_t err = suit_dfu_partition_device_info_get(&device_info);
+		if (req.image == 0) {
+			suit_plat_err_t err = suit_dfu_partition_device_info_get(&device_info);
 
-		if (err != SUIT_PLAT_SUCCESS) {
-			LOG_ERR("DFU Partition not found");
+			if (err != SUIT_PLAT_SUCCESS) {
+				LOG_ERR("DFU Partition not found");
+				return MGMT_ERR_ENOENT;
```

Comment:
these sorts of errors really should not be used, SMP version 2 should be used with dedicated error codes, see https://docs.zephyrproject.org/latest/services/device_mgmt/smp_protocol.html#minimal-response

Reply:
Thank you for your review. Yes, we are aware of that. This PR is an extension of already existing functionality and covers a fraction of SUIT SMP commands set. One day we will migrate all error codes (which will also require an adjustment in external tools).